library/core/src/time.rs

```diff
@@ -308,6 +307,35 @@ impl Duration {
         Duration { secs, nanos: subsec_nanos }
     }
 
+    /// Creates a new Duration from the specified number of nanoseconds.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the given number of weeks overflows the `Duration` size.
+    /// Use this function if you need to specify time greater than what can fit in u64
+    /// (around 584 years).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(duration_from_nanos_u128)]
+    /// use std::time::Duration;
+    /// let time_in_nanos = 2u128.pow(64);
+    /// let duration = Duration::from_nanos_u128(time_in_nanos);
+    /// ```
+    #[unstable(feature = "duration_from_nanos_u128", issue = "139201")]
+    #[must_use]
+    #[inline]
+    pub const fn from_nanos_u128(nanos: u128) -> Duration {
+        const NANOS_PER_SEC: u128 = self::NANOS_PER_SEC as u128;
+        let secs: u64 = (nanos / NANOS_PER_SEC) as u64;
```

Comment:
The function documentation currently states that the function panics on overflow,
but this cast truncates the result of the division to u64 instead.

e.g. `u128::MAX` results in `13088917067439035463.768211455s` instead of panicking.

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=a7dd4f02019a6076d496aab832128686)